dev/source/docs/building-setup-windows10.rst

```diff
@@ -16,31 +16,105 @@ Upon completion, you will be able to build ArduPilot binaries and run the native
+
+    * To reload the path variables in WSL either close the terminal and reopen it or use: ``logout``
+
+#. Install an XWindows application to run graphical programs such as SITL by installing `VcXsrv <https://sourceforge.net/projects/vcxsrv/>`_, `Cygwin X <https://x.cygwin.com/>`_ or `Xming <https://sourceforge.net/projects/xming/>`_ on Windows.
+
+#. Add the following in WSL at the end of ``~/.bashrc``
+
+.. code-block:: bash
+
+    # Export Display for XWindows for WSL2
+    export DISPLAY=$(grep -m 1 nameserver /etc/resolv.conf | awk '{print $2}'):0
+
+#. If desired, change default WSL parameters as described `at this link <https://docs.microsoft.com/en-us/windows/wsl/wsl-config#configure-global-options-with-wslconfig>`_. These parameters allow you to set for example the amount of memory and number of processors used by WSL.
+
+    .. note::
+
+        It is recommended to set memory limits as WSL2 currently does not release memory back to Windows properly.
+
+    * Create or edit the file at the following location according to your needs and system specifications, ``C:\Users\<yourUserName>\.wslconfig``
+
+    .. code-block:: bash
+
+        [wsl2]
+        memory=8GB # Limits VM memory in WSL to 8 GB
+        processors=16 # Makes the WSL 2 VM use 8 virtual processors
+        swap=8GB
+
+Using SITL with WSL and XWindows
+--------------------------------
+When using SITL with default options or ``--console --map`` the installed XWindows application must be started first. If not, the following error message will appear when running :
 
-  You can run XWindows applications (graphical programs, including SITL) by installing `VcXsrv <https://sourceforge.net/projects/vcxsrv/>`_, `Cygwin X <https://x.cygwin.com/>`_ or `Xming <https://sourceforge.net/projects/xming/>`_ on Windows. You will need to set the DISPLAY variable to point to your display: in WSL it is 127.0.0.1:0, in WSL2 it will be the ip address of the PC's network port and can be set by using the code below. For more information regarding configuring your XLaunch Software , you can refer to `this resource <https://itservices.cas.unt.edu/software/windows-x-server-vcxsrv>`_. Also in WSL2 you may have to disable the firewall for public networks, or create an exception in order for VcXsrv to communicate with WSL2:
-                                                                          
 ::
 
-  export DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0
+    [Errno 111] Connection refused sleeping
+
+Using VcXsrv
+++++++++++++
+#. Select display settings: "Multiple Windows" but it is your preference
+#. Select how to start clients: **Must** select **"Start no Client"**
+#. Extra settings: **Must** check **"Disable access control"**
+
+Code Editing in WSL Using VSCode
+---------------------------------------------
```

Comment:
underlines must match title width

Reply:
hmm it doesnt complain when it is too long, but sure does if it is too short :)